packages/test/test-end-to-end-tests/src/test/gcSummaryTests.spec.ts

```diff
@@ -0,0 +1,174 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+    ContainerRuntimeFactoryWithDefaultDataStore,
+    DataObject,
+    DataObjectFactory,
+} from "@fluidframework/aqueduct";
+import { assert, TelemetryNullLogger } from "@fluidframework/common-utils";
+import { IContainer } from "@fluidframework/container-definitions";
+import { ContainerRuntime } from "@fluidframework/container-runtime";
+import { Container } from "@fluidframework/container-loader";
+import { IFluidCodeDetails } from "@fluidframework/core-interfaces";
+import { SummaryType } from "@fluidframework/protocol-definitions";
+import { requestFluidObject } from "@fluidframework/runtime-utils";
+import { ITestDriver } from "@fluidframework/test-driver-definitions";
+import {
+    createAndAttachContainer,
+    createDocumentId,
+    createLoader,
+    OpProcessingController,
+} from "@fluidframework/test-utils";
+
+class TestDataObject extends DataObject {
+    public get _root() {
+        return this.root;
+    }
+
+    public get _context() {
+        return this.context;
+    }
+}
+
+describe("GC in summary", () => {
+    let documentId: string;
+    const codeDetails: IFluidCodeDetails = {
+        package: "garbageCollectionTestPackage",
+        config: {},
+    };
+    const dataObjectFactory = new DataObjectFactory(
+        "TestDataObject",
+        TestDataObject,
+        [],
+        []);
+    const runtimeOptions = {
+        generateSummaries: false,
+    };
+    const runtimeFactory = new ContainerRuntimeFactoryWithDefaultDataStore(
+        dataObjectFactory,
+        [
+            [dataObjectFactory.type, Promise.resolve(dataObjectFactory)],
+        ],
+        undefined,
+        undefined,
+        runtimeOptions,
+    );
+
+    let driver: ITestDriver;
+    let opProcessingController: OpProcessingController;
+    let containerRuntime: ContainerRuntime;
+    let defaultDataStore: TestDataObject;
+
+    async function createContainer(): Promise<IContainer> {
+        const loader = createLoader(
+            [[codeDetails, runtimeFactory]],
+            driver.createDocumentServiceFactory(),
+            driver.createUrlResolver());
+        return createAndAttachContainer(
+            codeDetails, loader, driver.createCreateNewRequest(documentId));
+    }
+
+    // Summarizes the container and validates that the data store's reference state is correct in the summary.
+    async function validateDataStoreReferenceState(dataStoreId: string, unreferenced?: true) {
```

Comment:
nit: I feel like `unreferenced` shouldn't be optional, so it's clear what the expectation is at the call site.

Reply:
Good point. I think making this `referenced` and a boolean would make the intention much clearer.